src/app/utils/CachedAxios.js

```diff
@@ -0,0 +1,20 @@
+import axios from 'axios';
+
+function CachedAxios() {
+  this.cache = {};
+  this.call = (url) => {
+    if (this.cache[url]) {
+      return new Promise((resolve) => {
+        resolve(this.cache[url]);
+      });
```

Comment:
This can just be `return Promise.resolve(this.cache[url])`. No need to create a `new Promise` if you can immediately return the resolved value.